### DISCUSSION TITLE
trouble and telescope workflow

### DISCUSSION THREAD

**Discussion by b1nhack** [folke/trouble.nvim#432] [General]
I noticed that trouble and telescope have some same use cases, such as lsp_definitions.
So, what is the recommended workflow, and should I use trouble to jump to lsp_definitions?
This way telescope has more segmented responsibilities and is only used for fuzzy search of data.

**Comment by pannapudi** [comment]
I've only found one integration between two plugins. It redirect diagnostics from telescope window into trouble [link to config](https://github.com/pudnax/configs/blob/master/.config/nvim/lua/plugins/telescope.lua#L81)
But it doesn't work if I ask Trouble to show diagnostics first.

**Reply by b1nhack** [reply]
Trouble can directly display the diagnostics
https://github.com/b1nhack/nvim/blob/aa4e920cb7f0b76f63ef00c0fbc0e11839531334/lua/plugins/trouble.lua#L66

**Comment by b1nhack** [comment]
Currently I use Trouble for a lot of lsp operations, Telescope is only used as a Fuzzy finder.
Especially after the https://github.com/folke/trouble.nvim/issues/450 fix.
In my opinion, Telescope is good for finding what you want from a large amount of data.
Trouble has a better interface to display lsp_definitions and so on.

**Comment by folke** [comment]
If I want to go somewhere, I use Telescope.
If I need to review a lot of locations/diagnostics I use Trouble.

I also frequently do searches with Telescope that I then open in Trouble to go through one by one.

**Reply by b1nhack** [reply]
The third item explains a use case for integration with Telescope.
I found a bunch of things I wanted, and I need to deal with them one by one.
